vagrant/ansible/roles/samba-glusterfs.setup/tasks/main.yml

```diff
@@ -0,0 +1,76 @@
+---
+    - name: Install samba and samba-vfs-glusterfs packages
+      yum:
+        name:
+          - samba
+          - samba-vfs-glusterfs
+        state: present
+
+    - name: Enable firewall for samba
+      firewalld:
+        service: samba
+        permanent: yes
+        state: enabled
+
+    - name: Install libsemanage-python. This is needed for the seboolean ansible command
+      yum: name=libsemanage-python state=present
+
+    - name: Selinux - Allow Samba to access glusterfs services
+      seboolean:
+        name: samba_load_libgfapi
+        state: yes
+        persistent: yes
+
+    - name: Copy over smb.conf
+      copy:
+        src: smb.conf
+        dest: /etc/samba/smb.conf
+        owner: root
+        group: root
+        mode: 0644
+
+    - name: Set Volume options and restart volume in a single host
+      block:
+      - name: Setup Gluster volume settings
+        gluster_volume:
+          state: present
+          name: "{{ gluster_cluster_volume }}"
+          options:
+            {
+              server.allow-insecure: 'on',
+              performance.cache-samba-metadata: 'on',
+              storage.batch-fsync-delay-usec: '0',
+              user.smb: 'on'
+            }
+      #Stop and Start gluster volume to run samba hooks to modify smb.conf
```

Comment:
Why do we need a restart? GlusterFS volume share section should get automatically added to _smb.conf_ when either _user.smb_ or _user.cifs_ volume set option is enabled for that volume in started state.

Reply:
done